scripts/install/install_unix.sh

```diff
@@ -582,6 +582,12 @@ install_package()
   fi
 
   info "Installing package..."
+  # if target install directory doesn't exist and we're using dpkg ensure a clean state 
+  # by checking for the package and running purge if it exists.
+  if [ ! -d "/opt/observiq-otel-collector" ] && [ "$package_type" = "deb" ]; then
+    dpkg -s "observiq-otel-collector" > /dev/null 2>&1 && dpkg --purge "observiq-otel-collector" > /dev/null 2>&1
+  fi
```

Comment:
I don't know how crazy we want to get with this, but there are other scenarios where the same thing can happen.
E.g. `rm /opt/observiq-otel-collector/config.yaml`, and now I don't have a config. Same thing if you remove the systemd file, since it's also a "config" file.

Those are more unlikely scenarios, though.

Reply:
The removing the entire directory was the most common cases developers and users experienced on an uninstall. I think I'm of the mind we take a wider approach now and then can tune it if people run into issues.

Reply:
Sounds good.

Reply:
If we end up needing to handle specific cases that are outside of goreleaser / nfpm's ability, we can add logic to the pre / post scripts. The collector's scripts are fairly simple, bindplane's is a good example https://github.com/observIQ/bindplane-op/tree/release/v1.19.0/scripts/package